member-service/src/main/java/com/hedvig/memberservice/services/signing/zignsec/ZignSecBankIdService.kt

```diff
@@ -98,26 +99,29 @@ class ZignSecBankIdService(
                 }
 
                 if (user != null) {
-                    if (result.memberId != user.associatedMemberId.toLong()) {
+                    val userMemberId = user.associatedMemberId.toLong()
+                    if (result.memberId != userMemberId) {
                         logger.info("ZignSec auth completion: MemberID mismatch, inactivating ${result.memberId}")
-                        commandGateway.sendAndWait<Void>(InactivateMemberCommand(result.memberId))
-                        apiGatewayService.reassignMember(result.memberId, user.associatedMemberId.toLong())
+                        commandGateway.sendAndWait<Unit>(InactivateMemberCommand(result.memberId))
+                        apiGatewayService.reassignMember(result.memberId, userMemberId)
                     }
                     logger.info("ZignSec auth completion: Sending success command")
-                    commandGateway.sendAndWait<Void>(
+                    commandGateway.sendAndWait<Unit>(
                         ZignSecSuccessfulAuthenticationCommand(
-                            user.associatedMemberId.toLong(),
+                            userMemberId,
                             result.id,
                             result.ssn,
                             ZignSecAuthenticationMarket.fromAuthenticationMethod(result.authenticationMethod),
                             result.identity.firstName,
                             result.identity.lastName
                         )
                     )
+                    commandGateway.sendAndWait<Unit>(
+                        PopulateMemberFromLoginDataCommand(userMemberId, result.identity.firstName, result.identity.lastName)
```

Comment:
Maybe this should be named 
```suggestion
                        MaybePopulateMemberFromLoginDataCommand(userMemberId, result.identity.firstName, result.identity.lastName)
```
Since we don't always insert the data?

Reply:
I actually disagree though. The API of the command is to set the data - the fact that it avoids firing events if it doesn't need to I think is an implementation detail of the aggregate.